sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -2,25 +2,60 @@
 #
 # License: BSD 3 clause
 
+from functools import partial
+import itertools
 import re
+import warnings
+
 import numpy as np
+import scipy
 from numpy.testing import assert_allclose
 import pytest
-import warnings
+from scipy import linalg
+from scipy.optimize import minimize, root
 
 from sklearn.base import clone
+from sklearn._loss import HalfBinomialLoss
 from sklearn._loss.glm_distribution import TweedieDistribution
 from sklearn._loss.link import IdentityLink, LogLink
 
-from sklearn.datasets import make_regression
+from sklearn.datasets import make_low_rank_matrix, make_regression
+from sklearn.linear_model import (
+    GammaRegressor,
+    PoissonRegressor,
+    Ridge,
+    TweedieRegressor,
+)
 from sklearn.linear_model._glm import _GeneralizedLinearRegressor
-from sklearn.linear_model import TweedieRegressor, PoissonRegressor, GammaRegressor
-from sklearn.linear_model import Ridge
+from sklearn.linear_model._linear_loss import LinearModelLoss
 from sklearn.exceptions import ConvergenceWarning
 from sklearn.metrics import d2_tweedie_score
 from sklearn.model_selection import train_test_split
 
 
+SOLVERS = ["lbfgs"]
+
+
+class BinomialRegressor(_GeneralizedLinearRegressor):
+    def _get_loss(self):
+        return HalfBinomialLoss()
```

Comment:
Isn't it great: Logistic regression defined in only 3 lines of code!
This way, we can test logistic regression too, with the same solvers. Eventually, we might want to use certain solvers in `LogisticRegression`. This way, they are already tested.

Reply:
I agree that at some point in the future we will refactor our `LogisticRegression` class to reuse the generic work that was done in `_GeneralizedLinearRegressor` and simplify a lot the code base.

However this is a non trivial refactoring that I would rather not start before we converge on better testing of GLMs with improved solvers such as #23314 and friends.

Reply:
> However this is a non trivial refactoring

What do you suggest/prefer: Remove `BinomialRegressor`? It is 3 lines of code in a test file +1 line to add in the test parametrization. That is hardly a burden.

Reply:
Side note on the possibility to implement multinomial logistic regression in a similar manner (with a `MultinomialRegressor` class): this could be a little awkward because all regressors in scikit-learn are expected to output a `y_pred` with the same shape as `y_train`, so `y_train` would have to be passed as one-hot codes which is not very memory or computationally efficient to treat multinomial logistic regression as a regressor instead of a classifier with hard predictions in `predict` and soft predictions in `predict_proba`.

Not however that I wouldn't be opposed to include a `MultinomialRegressor` in this test suite but that would require a slightly more complex fixture.

Reply:
Multinomial regression is a much harder problem when one tries to implement it efficiently. The important bits of `LinearModelLoss` are already used in `LogisticRegression` and tested there. It would add extensively more code here.